### ISSUE TITLE
is pumvisible()? surpported?

### ISSUE THREAD

**Issue by lookforit** [jaredpar/VsVim#1402]
I just want to define a specific map  when pop-up menu appears:

``` viml
inoremap <expr> <c-k> pumvisible()?"\<Up>":"\<c-k>"  
inoremap <expr> <c-j> pumvisible()?"\<Down>":"\<c-j>"  
```

So I  wonder is pumvisible() function surpported by `vsvim`?

**Comment by sfinktah** [comment]
It's a pretty common function, a quick `ack` of my .vim gives me quite a list.  But in terms of implementation, it simply returns 0 unless there is a pop-up menu active (completion, what-not).

> You can use the pumvisible() function to test if the popup menu is visible, it returns 0 when it's not, 1 otherwise.
> Then you can build an expression with the ternary operator ?: (see :h expr1 for more info) to return a different string depending on the state of the popup menu. For example:
> `pumvisible() ? '<C-N>' : '<C-O>gj'`

I wouldn't normally care, but evidently something is bound in my .vimrc such that when I press Enter in insert mode, I get a free `pumvisible() ? "\"`.

```
bundle/disabled/vim-nodejs-complete/README.md
bundle/emmet-vim/autoload/emmet/util.vim
bundle/ultisnips/autoload/UltiSnips.vim
bundle/ultisnips/doc/UltiSnips.txt
bundle/unite.vim/autoload/unite/mappings.vim
bundle/YouCompleteMe/autoload/youcompleteme.vim
bundle.snips/ultisnips/autoload/UltiSnips.vim
bundle.snips/ultisnips/doc/UltiSnips.txt
disabled/snipmate.vim/plugin/snipMate.vim
ulti.vim
vim-addons/tlib/autoload/tlib/map.vim
vim-addons/tlib/doc/tlib.txt
vim-addons/vim-addon-manager/autoload/vam/utils.vim
```

**Comment by sfinktah** [comment]
BTW, I fixed my immediate problem. I had a script that was indirectly sourced from .vimrc with the line:
  
    inoremap <expr> <CR> pumvisible() ? "\<C-y>" : "\<C-g>u\<CR>"

Something I added myself.  If it had been one of the many other instances that was within a plugin, I guess it wouldn't have loaded and I wouldn't have had the issue.

BTW, very impressed with the level of detail.  Supported some rather unexpected features such as <Leader><Space> to deselect highlighted search items.  Unless that is one I added by hand to .vimrc, either way, impressive.
